test/gtest/ucp/test_ucp_request.cc

```diff
@@ -0,0 +1,113 @@
+/**
+ * Copyright (C) Mellanox Technologies Ltd. 2021.  ALL RIGHTS RESERVED.
+ *
+ * See file LICENSE for terms.
+ */
+#include <string.h>
+#include "ucp_test.h"
+#include <common/mem_buffer.h>
+extern "C" {
+#include <ucp/core/ucp_worker.h>
+}
+
+
+class test_ucp_request : public ucp_test {
+public:
+    virtual void init()
+    {
+        ucp_test::init();
+        sender().connect(&receiver(), get_ep_params());
+        int mem_type_pair_index = get_variant_value() % mem_type_pairs.size();
+        m_send_mem_type  = mem_type_pairs[mem_type_pair_index][0];
+        m_recv_mem_type  = mem_type_pairs[mem_type_pair_index][1];
+    }
+
+    static void get_test_variants(std::vector<ucp_test_variant> &variants)
+    {
+        int count = 0;
+        add_variant(variants, UCP_FEATURE_TAG);
+        for (std::vector<std::vector<ucs_memory_type_t> >::const_iterator iter =
+                 mem_type_pairs.begin(); iter != mem_type_pairs.end(); ++iter) {
+                std::string name =
+                        std::string(ucs_memory_type_names[(*iter)[0]]) + ":" +
+                        std::string(ucs_memory_type_names[(*iter)[1]]);
+                add_variant_with_value(variants, UCP_FEATURE_TAG, count, name);
+                ++count;
```

Comment:
1. indent
2. add memtype to variant name, to not print with UCS_TEST_MESSAGE (AFAIR there is variant for memtype bitmap)
3. can enumerate on one dimension of memtype (and check both send/recv reqs) since query function is one sided